### ISSUE TITLE
Formatting check on input names

### ISSUE THREAD

**Issue by nickjcroucher** [johnlees/PopPUNK#46]
When processing all the way to microreact, any file name with more than one '.' will fail, because different amounts of the filename will be treated as a suffix and lost.

When using rapidnj, underscores are converted to spaces.

Should we convert these characters to dashes automatically, or warn the user to change the names themselves?

**Comment by johnlees** [comment]
For microreact, is it an issue with(/feature of) the way microreact processes these names, or an issue in our code when trying to infer the suffix?

For rapidnj, as we read this output in with dendropy anyway, I think the best solution would be to turn these spaces back into underscores ourselves, using the known sample names.

**Comment by bede** [comment]
Is the following error related?

```
Writing microreact output
Building phylogeny
Traceback (most recent call last):
  File "/well/bag/uzv018/bin/conda/envs/poppunk/bin/poppunk", line 10, in <module>
    sys.exit(main())
  File "/well/bag/uzv018/bin/conda/envs/poppunk/lib/python3.7/site-packages/PopPUNK/__main__.py", line 350, in main
    args.output, args.info_csv, args.rapidnj, overwrite = args.overwrite)
  File "/well/bag/uzv018/bin/conda/envs/poppunk/lib/python3.7/site-packages/PopPUNK/plot.py", line 631, in outputsForMicroreact
    generate_phylogeny(coreMat, seqLabels, outPrefix, "_core_NJ.nwk", rapidnj, overwrite)
  File "/well/bag/uzv018/bin/conda/envs/poppunk/lib/python3.7/site-packages/PopPUNK/plot.py", line 670, in generate_phylogeny
    is_first_column_row_names=False)
  File "/well/bag/uzv018/bin/conda/envs/poppunk/lib/python3.7/site-packages/dendropy/calculate/phylogeneticdistance.py", line 179, in from_csv
    **csv_reader_kwargs
  File "/well/bag/uzv018/bin/conda/envs/poppunk/lib/python3.7/site-packages/dendropy/utility/container.py", line 747, in from_csv
    **csv_reader_kwargs)
  File "/well/bag/uzv018/bin/conda/envs/poppunk/lib/python3.7/site-packages/dendropy/utility/container.py", line 790, in _from_csv_file
    data_type=column_data_types.get(cell, default_data_type))
  File "/well/bag/uzv018/bin/conda/envs/poppunk/lib/python3.7/site-packages/dendropy/utility/container.py", line 815, in add_column
    assert column_name not in self._column_name_set
AssertionError
```

Is the suggested workound for this issue to `sed` the names in the *.msh and *.pkl files to not contain special characters?

My sample names don't contain multiple `.` but do contain `/` and `-`

**Comment by johnlees** [comment]
Could you try running without rapidNJ to see if that is what's causing the problem? What are some of the sample names?

**Comment by bede** [comment]
Ok, after running `mash triangle` on the sketches, importing the matrix into python in order convert lower triangle phylip to square phylip format, it looks like RapidNJ doesn't mind the names, which are in the following format, are 96 chars long and not duplicated. I guess it's something else.

`/absolute/path/to/assembly/cf206395-2d5e-43a0-8343-7a91f15bec38/contigs.fa`

It would be great if there were a way to specify the sample name in the `--create-db` step, or at least alter them afterwards without resorting to painfully loading, modifying and dumping pickles and numpy arrays

**Comment by bede** [comment]
FTR, the issue was using path tails with the same name (`contigs.fa`) as input, albeit having different paths. I symlinked these to `{id}.fa` and all is well.

**Comment by johnlees** [comment]
Closed in 620aa4a7f63b01c97f8cbd66f7bee1e7d314b2bb
